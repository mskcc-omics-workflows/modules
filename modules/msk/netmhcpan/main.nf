process NETMHCPAN {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/netmhctools:1.0.0':
        'docker.io/mskcc/netmhctools:1.0.0' }"

    input:
    tuple val(meta),  path(inputFasta), val(hlaString), val(inputType)

    output:
    tuple val(output_meta),       path("*.xls"),               emit: xls
    tuple val(output_meta),       path("*.netmhcpan.output"),  emit: netmhcpanoutput
    path "versions.yml",                                       emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def hla = hlaString.trim()
    output_meta = meta.clone()
    output_meta.typeMut = inputType == "MUT" ? true : false
    output_meta.fromStab = false

    """
    /usr/local/bin/netMHCpan-4.1/netMHCpan -s 1 -BA 1 -f ${inputFasta} -a ${hla} -l 9,10 -inptype 0 -xls -xlsfile ${prefix}.${inputType}.xls > ${prefix}.${inputType}.netmhcpan.output


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        netmhcpan: v4.1
    END_VERSIONS

    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    output_meta = meta.clone()
    output_meta.typeMut = inputType == "MUT" ? true : false
    output_meta.fromStab = false
    """
    touch ${prefix}.MUT.netmhcpan.output
    touch ${prefix}.MUT.xls

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        netmhcpan: v4.1
    END_VERSIONS
    """
}
