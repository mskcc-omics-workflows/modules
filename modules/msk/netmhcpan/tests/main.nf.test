nextflow_process {

    name "Test Process NETMHCPAN"
    script "../main.nf"
    process "NETMHCPAN"

    tag "modules"
    tag "modules_nfcore"
    tag "netmhcpan"
    tag "modules_msk"

    test("netmhcpan - MUT - xls,output,fa") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(file(params.test_data_mskcc['neoantigen']['MUT_sequence_fa']), checkIfExists: true),
                    "HLA-A24:02,HLA-A24:02,HLA-B39:01,HLA-B39:01,HLA-C07:01,HLA-C06:02",
                    "MUT"
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.netmhcpanoutput[0][0],
                    file(process.out.xls[0][1]).name,
                    file(process.out.netmhcpanoutput[0][1]).name
                    ).match()
                }
            )
        }

    }

    test("netmhcpan - WT - xls,output,fa") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(file(params.test_data_mskcc['neoantigen']['WT_sequence_fa']), checkIfExists: true),
                    "HLA-A24:02,HLA-A24:02,HLA-B39:01,HLA-B39:01,HLA-C07:01,HLA-C06:02",
                    "WT"
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.netmhcpanoutput[0][0],
                    file(process.out.xls[0][1]).name,
                    file(process.out.netmhcpanoutput[0][1]).name
                    ).match()
                }
            )
        }

    }



    test("netmhcpan - xls,output,fa - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file('MUT_sequence_fa'),
                    "HLA",
                    "MUT"
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.netmhcpanoutput[0][0],
                    file(process.out.xls[0][1]).name,
                    file(process.out.netmhcpanoutput[0][1]).name
                    ).match()
                }
            )
        }

    }

}
