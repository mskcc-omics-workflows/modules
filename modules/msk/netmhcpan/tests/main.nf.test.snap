{
    "netmhcpan - WT - xls,output,fa": {
        "content": [
            [
                "versions.yml:md5,35ec563839ee27410cf9f8d134c6e8e5"
            ],
            {
                "id": "test",
                "single_end": false,
                "typeMut": false,
                "fromStab": false
            },
            "test.WT.xls",
            "test.WT.netmhcpan.output"
        ],
        "timestamp": "2024-05-03T13:49:45.932532"
    },
    "netmhcpan - xls,output,fa - stub": {
        "content": [
            [
                "versions.yml:md5,35ec563839ee27410cf9f8d134c6e8e5"
            ],
            {
                "id": "test",
                "single_end": false,
                "typeMut": true,
                "fromStab": false
            },
            "test.MUT.xls",
            "test.MUT.netmhcpan.output"
        ],
        "timestamp": "2024-05-03T13:49:54.704381"
    },
    "netmhcpan - MUT - xls,output,fa": {
        "content": [
            [
                "versions.yml:md5,35ec563839ee27410cf9f8d134c6e8e5"
            ],
            {
                "id": "test",
                "single_end": false,
                "typeMut": true,
                "fromStab": false
            },
            "test.MUT.xls",
            "test.MUT.netmhcpan.output"
        ],
        "timestamp": "2024-05-03T13:49:33.972805"
    }
}